tests/cupy_tests/lib_tests/test_polynomial.py

```diff
@@ -272,3 +304,97 @@ def test_poly1d_ne1(self, xp, dtype):
     def test_poly1d_ne2(self, xp, dtype):
         a, b = self.make_poly1d2(xp, dtype)
         return a != b
+
+
+@testing.gpu
+@testing.parameterize(*testing.product({
+    'shape1': [(), (0,), (3,), (5,)],
+    'shape2': [(), (0,), (3,), (5,)]
+}))
+class TestPolyaddShapeCombination(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_polyadd(self, xp, dtype):
+        a = testing.shaped_arange(self.shape1, xp, dtype)
+        b = testing.shaped_arange(self.shape2, xp, dtype)
+        with cupyx.allow_synchronize(False):
+            return xp.polyadd(a, b)
+
+
+@testing.gpu
+class TestPolyadd(unittest.TestCase):
```

Comment:
Why did you split the tests into `TestPoly1dArithmetic` and `TestPolyAdd`?

Reply:
I didn't split tests. `TestPoly1dArithmetic` class tests __add__  and `TestPolyAdd` class tests `polyadd`. I know that they are the same in the `addition` case apart from the fact that I call `polyadd` directly or use `+`. Would you prefer to eliminate `TestPoly1dArithmetic` ?

Reply:
OK. Let's leave `TestPoly1dArithmetic` without eliminating.

Reply:
For the refactoring of the tests, let's organize which tests are needed.
- The type of inputs/outputs
  - 16 types of input combination: (poly1d, array, numpy scalar, or python scalar) v.s. (same as on the left)
  - should be placed on `TestPoly1dArithmetic`.
- Device synchronization
  - should be checked at the same time as the type checking.
- Shape combinations
- Dtype combinations
- Leading zeros
  - I think we don't need to check it because `polyadd` does not trim the leading zeros. It is already tested in `TestPoly1d`.

Reply:
Concerning the 16 combinations, two of them fail

1. numpy scalar + `poly1d` because of Implicit conversion to NumPy array
2. array + `poly1d` because `poly1d` type is not supported in `ndarray.__add__`

Reply:
`test_polyadd_ndim` is not in the list. So will be eliminated right?

Reply:
Yes, it's OK to remove because NumPy does not guarantee to raise an error for such inputs.